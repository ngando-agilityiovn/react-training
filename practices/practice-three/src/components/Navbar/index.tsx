import { Flex, ListItem, UnorderedList } from '@chakra-ui/react'
import { Link } from 'react-router-dom'

interface INavbarItem {
  title: string
  link: string
}

interface INavbar {
  data: INavbarItem[]
}

const Navbar = ({ data }: INavbar) => {
  return (
    <UnorderedList listStyleType="none">
      <Flex>
        {data.map(({ title, link }) => (
          <ListItem key={title} mr="50px" color="secondary" fontWeight="medium">
            <Link to={link}>{title}</Link>
          </ListItem>
        ))}
      </Flex>
    </UnorderedList>
  )
}

export default Navbar
