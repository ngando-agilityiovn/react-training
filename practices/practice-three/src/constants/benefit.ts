export const BENEFIT_DATA = [
  {
    title:
      'Durable leather is easily cleanable so you can keep your look fresh.',
  },
  {
    title: 'Toe piece with star pattern adds durability.',
  },
]
