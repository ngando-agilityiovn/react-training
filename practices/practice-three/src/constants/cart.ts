export const CARD_DATA = [
  {
    id: '0',
    name: 'T-shirt',
    ratings: 5,
    price: 200,
    currency: '$',
    quantity: 2,
    reviews: [
      {
        id: 1,
        reviewer: 'John Doe',
        comments: 'Great product, very satisfied with my purchase!'
      },
      {
        id: 2,
        reviewer: 'Jane Smith',
        comments: 'Excellent service and fast delivery.'
      },
      {
        id: 3,
        reviewer: 'Bob Johnson',
        comments: 'Product was okay, but could be improved in certain areas.'
      }
    ],
    images: [
      'https://source.unsplash.com/random/587x691/?fashion',
      'https://source.unsplash.com/random/587x691/?clothes',
      'https://source.unsplash.com/random/587x691/?t-shirt',
      'https://source.unsplash.com/random/587x691/?adidas'
    ]
  },
  {
    id: '1',
    name: 'T-shirt',
    ratings: 28,
    price: 340,
    currency: '$',
    quantity: 20,
    reviews: [
      {
        id: 1,
        reviewer: 'John Doe',
        comments: 'Great product, very satisfied with my purchase!'
      },
      {
        id: 2,
        reviewer: 'Jane Smith',
        comments: 'Excellent service and fast delivery.'
      },
      {
        id: 3,
        reviewer: 'Bob Johnson',
        comments: 'Product was okay, but could be improved in certain areas.'
      }
    ],
    images: [
      'https://source.unsplash.com/random/587x691/?fashion',
      'https://source.unsplash.com/random/587x691/?clothes',
      'https://source.unsplash.com/random/587x691/?t-shirt',
      'https://source.unsplash.com/random/587x691/?adidas'
    ]
  }
]
