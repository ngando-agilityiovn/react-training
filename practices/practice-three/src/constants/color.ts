export const DATA_COLOR = [
  '#ffd3d3',
  '#ffd3f8',
  '#bbd278',
  '#ecdecc',
  '#bbc1f8'
]
