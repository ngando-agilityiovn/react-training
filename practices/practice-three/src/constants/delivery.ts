export const DELIVERY_DATA = [
  {
    title: 'Free Delivery',
    description: 'Enter your Postal code for Delivery Availability',
  },
  {
    title: 'Return Delivery',
    description: 'Free 30 days Delivery Return. Details',
  },
]
