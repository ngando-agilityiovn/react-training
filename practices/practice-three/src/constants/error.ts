export const NOTIFY = {
  ADD_FAILD: 'Adding failed products'
}
