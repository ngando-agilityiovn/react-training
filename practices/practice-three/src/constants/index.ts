export * from './navbar'
export * from './filter'
