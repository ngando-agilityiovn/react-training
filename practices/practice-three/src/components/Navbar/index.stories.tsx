import { Meta, StoryObj } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

// Constants
import { NAVBAR } from '@/constants'

// Components
import Navbar from '.'

export default {
  title: 'Components/Navbar',
  component: Navbar,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
} as Meta

type Story = StoryObj<typeof Navbar>

export const Default: Story = {
  args: {
    data: NAVBAR,
  },
}
