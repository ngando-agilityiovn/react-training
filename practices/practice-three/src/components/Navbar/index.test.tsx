import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

// Constants
import { NAVBAR } from '../../constants'

// Components
import Navbar from '.'

describe('Navbar component', () => {
  it('Render correcty', () => {
    const container = render(
      <BrowserRouter>
        <Navbar navbarData={NAVBAR} />
      </BrowserRouter>,
    )
    expect(container).toMatchSnapshot()
  })
})
