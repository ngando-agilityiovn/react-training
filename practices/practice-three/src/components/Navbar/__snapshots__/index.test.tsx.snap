// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Render correcty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="css-1laduhn"
        role="list"
      >
        <div
          class="css-k008qs"
        >
          <li
            class="css-1a25oy"
          >
            <a
              href="/category"
            >
              Category
            </a>
          </li>
          <li
            class="css-1a25oy"
          >
            <a
              href="/brand"
            >
              Brand
            </a>
          </li>
          <li
            class="css-1a25oy"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="css-1a25oy"
          >
            <a
              href="/faq"
            >
              FAQ’s
            </a>
          </li>
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="css-1laduhn"
      role="list"
    >
      <div
        class="css-k008qs"
      >
        <li
          class="css-1a25oy"
        >
          <a
            href="/category"
          >
            Category
          </a>
        </li>
        <li
          class="css-1a25oy"
        >
          <a
            href="/brand"
          >
            Brand
          </a>
        </li>
        <li
          class="css-1a25oy"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="css-1a25oy"
        >
          <a
            href="/faq"
          >
            FAQ’s
          </a>
        </li>
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
